apps/designer/app/canvas/canvas.tsx

```diff
@@ -174,6 +175,11 @@ export const Canvas = ({ data }: CanvasProps): JSX.Element | null => {
   useInitializeDesignTokens(data.designTokens);
   const assets = useAssets(data.assets);
   useAllUserProps(data.props);
+  useState(() => {
+    if (data.tree?.presetStyles) {
+      presetStylesContainer.value = data.tree.presetStyles;
+    }
+  });
```

Comment:
this should be probably a custom hook with a semantic name, that goes along with other things for presetStyles on canvas

Reply:
Or we could use single useState to populate data in atoms

Reply:
yeah, though keeping concerns in the specific modules makes it more maintainable long-term, downside - you have to jump into that file when reading

I think its generally a downside with organizing things well, the better it is organized, the more you have to jump, its a tradeoff for sure, but I don't think we can scale without organizing things semantically.

Inline stuff like this will grow in 3-6 month into a monster.